community/kernel/src/test/java/org/neo4j/kernel/impl/api/KernelTransactionsTest.java

```diff
@@ -413,12 +414,62 @@ public void exceptionWhenStartingNewTransactionOnShutdownInstance() throws Throw
         SecurityContext securityContext = mock( SecurityContext.class );
 
         availabilityGuard.shutdown();
-        Executors.newSingleThreadExecutor().submit( () -> stopKernelTransactions( kernelTransactions ) ).get();
+        t2.execute( (OtherThreadExecutor.WorkerCommand<Void,Void>) state ->
+        {
+            stopKernelTransactions( kernelTransactions );
+            return null;
+        } );
 
         expectedException.expect( DatabaseShutdownException.class );
         kernelTransactions.newInstance( KernelTransaction.Type.explicit, securityContext, 0L );
```

Comment:
and this is miserable mistake! was thinking for some reason that OtherThreadRule doing that for me, will clean this up!